ibis/backends/duckdb/__init__.py

```diff
@@ -1255,3 +1322,16 @@ def _insert_dataframe(
             if overwrite:
                 con.execute(t.delete())
             con.execute(t.insert().from_select(columns, sa.select(source)))
+
+    def table(
+        self,
+        name: str,
+        database: str | None = None,
+        schema: str | None = None,
+    ) -> ir.Table:
+        expr = super().table(name=name, database=database, schema=schema)
+        # load geospatial only if geo columns
+        if any(typ.is_geospatial() for typ in expr.op().schema.types):
```

Comment:
If, at some point in the future, this starts to become a performance bottleneck, we can instead run this check in `SQL` and it will be faster (but not necessary to handle that now).